src/utils/browser.js

```diff
@@ -21,5 +21,17 @@ export default {
             let m = num.toExponential().match(/\d(?:\.(\d*))?e([+-]\d+)/)
             return num.toFixed(Math.max(0, (m[1] || '').length - m[2]))
         }
+    },
+    getLength(str) {
+        var n = str.length,s
+        var len=0
+        for(var i=0; i <n;i++){
+            s=str.charCodeAt(i);
+            while( s > 0 ){
+                len++;
+                s = s >> 8;
```

Comment:
the string encode on chain is using UTF-8, so every non-ascii char will be length of 3.